corehq/apps/users/models.py

```diff
@@ -1365,53 +1422,74 @@ def supply_point_index_mapping(self, supply_point, clear=False):
             )
 
     def add_location(self, location, create_sp_if_missing=False):
+        """
+        Add a single location to the delgate case access.
+        """
         from corehq.apps.commtrack.models import SupplyPointCase
 
         sp = SupplyPointCase.get_or_create_by_location(location)
 
-        from corehq.apps.commtrack.util import submit_mapping_case_block
-        submit_mapping_case_block(self, self.supply_point_index_mapping(sp))
+        if not location.location_type_object.administrative:
+            from corehq.apps.commtrack.util import submit_mapping_case_block
+            submit_mapping_case_block(self, self.supply_point_index_mapping(sp))
 
     def submit_location_block(self, caseblock):
         from corehq.apps.hqcase.utils import submit_case_blocks
 
         submit_case_blocks(
-            ElementTree.tostring(caseblock.as_xml(format_datetime=json_format_datetime)),
+            ElementTree.tostring(
+                caseblock.as_xml(format_datetime=json_format_datetime)
+            ),
             self.domain,
             self.username,
             self._id
         )
 
     def remove_location(self, location):
+        """
+        Remove a single location from the case delagate access.
+        """
         from corehq.apps.commtrack.models import SupplyPointCase
 
         sp = SupplyPointCase.get_by_location(location)
 
         mapping = self.get_location_map_case()
 
-        if mapping and location._id in [loc._id for loc in self.locations]:
-            caseblock = CaseBlock(
-                create=False,
-                case_id=mapping._id,
-                version=V2,
-                index=self.supply_point_index_mapping(sp, True)
-            )
+        if not location.location_type_object.administrative:
+            if mapping and location._id in [loc._id for loc in self.locations]:
+                caseblock = CaseBlock(
+                    create=False,
+                    case_id=mapping._id,
+                    version=V2,
+                    index=self.supply_point_index_mapping(sp, True)
+                )
 
-            self.submit_location_block(caseblock)
+                self.submit_location_block(caseblock)
 
     def clear_locations(self):
+        """
+        Wipe all case delagate access.
+        """
         mapping = self.get_location_map_case()
         if mapping:
             mapping.delete()
 
-    def set_locations(self, locations):
+    def create_location_delegates(self, locations):
+        """
+        Submit the case blocks creating the delgate case access
+        for the location(s).
+        """
         from corehq.apps.commtrack.models import SupplyPointCase
```

Comment:
you could add an assert here saying that type should not be `administrative_only`